### PR TITLE
client: make set/delete one shot operations

### LINKED ISSUE #5832: atomic Compare-and-Swap PrevNoExist, the operation failed,but the key was stored
etcd:  v2.3.6

there is a demo [https://github.com/yorkart/etcd-demo](https://github.com/yorkart/etcd-demo)
when loop invoke atomic set and delete
after some times get error :

- set error (but  value of the key has stored in it)
` set key error:  105: Key already exists (/demo/a) [1533028]`

- delete (indeed key has been deleted )
`delete key error:  100: Key not found (/demo/a) [1530634]`

### PR DESCRIPTION
**Pull request by heyitsanthony**
Old behavior would retry set and delete even if there's an error. This
can lead to the client returning an error for deleting twice, instead
of returning an error for an interdeterminate state.

Fixes #5832

### CODE REVIEW

**Comment by xiang90** [comment]
is it easy to add a test for this?

**Comment by heyitsanthony** [comment]
@xiang90 I think so; will add some integration tests around it

**Comment by heyitsanthony** [comment]
tests added /cc @xiang90

**Comment by xiang90** [comment]
lgtm

File: integration/v2_client_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"github.com/coreos/etcd/client"
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+// TestV2NoRetryEOF tests destructive api calls won't retry on a disconnection.
+func TestV2NoRetryEOF(t *testing.T) {
```
Comment (xiang90) [inline integration/v2_client_test.go]: one thing: shall we move this to client/integration?

Comment (heyitsanthony) [inline integration/v2_client_test.go]: seems like overkill, maybe if we add more tests in the future?

**Comment by heyitsanthony** [comment]
moved to `client/integration`, exported some bits of `integration` /cc @xiang90

**Comment by xiang90** [comment]
lgtm

**Pull request merged**